src/js/reducers/pipelines.js

```diff
@@ -0,0 +1,74 @@
+/* eslint new-cap:0 */
+'use strict';
+
+var Immutable = require('immutable');
+
+var actions = require('../constants/actions');
+var counter = require('../util/counter');
+var immutableUtils = require('../util/immutable-utils');
+var get = immutableUtils.get;
+var getIn = immutableUtils.getIn;
+var set = immutableUtils.set;
+var setIn = immutableUtils.setIn;
+var ensureValuePresent = immutableUtils.ensureValuePresent;
+var ensureValueAbsent = immutableUtils.ensureValueAbsent;
+var assign = require('object-assign');
+
+
+/**
+ * Main pipelines reducer function, which generates a new state for the
+ * pipelines (pipelines) property store based on the changes specified by the
+ * dispatched action object.
+ *
+ * @param {Object} state - An Immutable.Map state object
+ * @param {Object} action - A redux action object
+ * @returns {Object} A new Immutable.Map with the changes specified by the action
+ */
+function pipelinesReducer(state, action) {
+  if (typeof state === 'undefined') {
+    return Immutable.fromJS({
+      pipelines: []
+    });
+  }
```

Comment:
Because we use [combineReducers](http://redux.js.org/docs/api/combineReducers.html) in `reducers/index.js`, the `state` that is received by this reducer only contains our pipelines state. So we don't need to further nest it in here. 

Also, rather than using an array to store the pipelines, lets switch to an `ImmutableMap`. The keys in this map will be the pipeline IDs, and saves us iterating through the array to find the pipeline we care about for other actions (e.g., your L50 codeblock). Take a look at the `marks` or `signals` reducers for an example.